base/uk.ac.stfc.isis.ibex.scriptgenerator/python_support/script_definition_loader.py

```diff
@@ -84,14 +93,41 @@ def getParameters(self) -> List[PythonActionParameter]:
 
         kwargs_with_defaults = []
 
+        enum_parameter_names = []
+        
+        try:
+            enum_parameter_names = getattr(self.script_definition,"enum_params")
+        except:
+            enum_parameter_names = []
```

Comment:
It doesn't really matter in this case due to how `hasattr()` is implemented, but its generally better practice to check for the presence of a value rather than trying to get it and catching an exception if its not there. i.e.
```
if hasattr(self.script_definition,"enum_params"):
    enum_parameter_names = getattr(self.script_definition,"enum_params")
else:
    enum_parameter_names = []
```

Reply:
Done